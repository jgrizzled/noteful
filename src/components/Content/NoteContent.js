import React from 'react';
import styled from 'styled-components';
import ContentSection from 'components/Content/ContentSection';
import NoteCard from 'components/Content/NoteCard';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Text = styled.p`
  line-height: 1.5rem;
  padding: 2rem;
  margin: 0;
`;

export default props => (
  <ContentSection>
    <Container>
      <NoteCard note={props.note} />
      <Text>{props.note.content}</Text>
    </Container>
  </ContentSection>
);
