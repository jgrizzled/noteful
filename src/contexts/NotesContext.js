import React from 'react';

export default React.createContext({
  notes: [],
  folders: [],
  addNote: () => null,
  deleteNote: () => null,
  addFolder: () => null
});
