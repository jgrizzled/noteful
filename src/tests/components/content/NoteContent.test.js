import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import NoteContent from 'components/content/NoteContent';
import NotesContext from 'contexts/NotesContext';

const notesContextValue = {
  notes: [
    {
      id: '0',
      name: 'test note',
      modified: '',
      content: '',
      folderId: '0'
    }
  ],
  folders: [
    {
      id: '0',
      name: 'test folder'
    }
  ],
  addNote: () => true,
  deleteNote: () => true,
  addFolder: () => true
};

const component = (
  <BrowserRouter>
    <NotesContext.Provider value={notesContextValue}>
      <NoteContent />
    </NotesContext.Provider>
  </BrowserRouter>
);

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(component, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
