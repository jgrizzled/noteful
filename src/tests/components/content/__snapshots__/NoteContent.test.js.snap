// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<h1
  className="sc-AykKF cfslnx"
>
  Note not found!
</h1>
`;
